e2e/etcd_test.go

```diff
@@ -424,6 +424,9 @@ func (epc *etcdProcessCluster) StopAll() (err error) {
 func (epc *etcdProcessCluster) Close() error {
 	err := epc.StopAll()
 	for _, p := range epc.procs {
+		if p == nil { // newEtcdProcess failed in the middle
```

Comment:
Can we add more comments? mention that this is for cleaning up tests when there is a failure in the middle?